code/modules/admin/fun_verbs.dm

```diff
@@ -1111,3 +1111,37 @@ GLOBAL_LIST_EMPTY(custom_outfits)
 
 	log_admin("[key_name(usr)] updated the appearance of [key_name(H)].")
 	message_admins("[ADMIN_TPMONTY(usr)] updated the appearance of [ADMIN_TPMONTY(H)].")
+
+
+/datum/admins/proc/offer(var/mob/M in GLOB.mob_list)
+	set category = "Fun"
+	set name = "Offer Mob"
+
+	if(!check_rights(R_FUN))
+		return
+
+	if(!isliving(M))
+		return
+
+	var/mob/living/L = M
+
+	if(L.key || L.ckey)
+		if(alert("This mob has a player inside, are you sure you want to proceed?", "WARNING", "Yes", "No") != "Yes")
+			return
+		L.taken = FALSE
+		var/mob/dead/observer/ghost = L.ghostize(FALSE)
+		if(ghost)
+			ghost.timeofdeath = world.time
+	else if(L.taken)
+		if(alert("This mob has been offered, are you sure you want to proceed?", "Warning", "Yes", "No") != "Yes")
+			return
+		L.taken = FALSE
+
+	if(alert("Are you sure?", "Offer Mob", "Yes", "No") != "Yes")
```

Comment:
Is this second alert necessary?

Reply:
It's a first alert technically and yeah a popup is necessary with this dangerous verb.